packages/@atjson/document/src/document.ts

```diff
@@ -403,12 +411,24 @@ export class Document {
   }
 
   /**
-   * Cuts out part of the document, modifying `this` and returning the removed portion
+   * Cuts out part of the document, modifying `this` and returning the removed portion.
+   * By default, the cut contains a truncated copy of any annotation
+   * overlapping the range. This can be overriden by specifying a filter.
+   * Annotations included wholly in the cut that are matched by the filter
+   * will be removed from the original document
    */
-  cut(start: number, end: number): Document {
-    let slice = this.slice(start, end);
+  cut(
+    start: number,
+    end: number,
+    filter?: (annotation: Annotation<any>) => boolean
+  ): Document {
+    let slice = this.slice(start, end, filter);
     this.where(function annotationWasCut(annotation) {
-      return annotation.start >= start && annotation.end <= end;
+      return (
+        annotation.start >= start &&
+        annotation.end <= end &&
+        (!filter || filter(annotation))
```

Comment:
hmm, is this the right behavior here? If you're cutting out part of a document but you only care about some of the annotations, do you want to leave any other annotations in the range collapsed in place?

Reply:
what if we did something more like the boundary behaviors, where we have two available behaviors: either the result document keeps external wrapping annotations or only keeps annotations that lie fully within the range. and then maybe separate behaviors for the original document if that becomes necessary?

Reply:
My reasoning for the filter behavior here is that right now `cut` does not drop any annotations: they are either copied to the new document or are retained in the original (or both). I'd like to keep that behavior here, as clients can then choose to drop annotations from whichever documents after performing `cut`.

As far as giving a set of predefined behaviors, I thought about that but decided against it for two reasons:
1. I think there are subtle differences in boundary behaviors that some people might actually want: do you include the ends of the range or not, do you want different behavior on the right vs left side of the range, etc. So there are actually in my mind more than 2 valid boundary behaviors and I don't want to try to enumerate all of them.
2. Boundary behaviors only allow you to consider the positions of annotations when deciding whether or not to include them. It's possible that there are more things you want to consider, particularly when annotations coincide. For example, in a canonicalized document where all of the text under parse tokens has been deleted, we might have annotations like so, all over the same range:
```
Section > List > ListItem > Photo > Caption > Paragraph > Bold
```
and if we wanted to cut out the caption in this case we'd need to look at other properties of the annotation - its type, rank, etc - in order to determine what to include in the cut.